test/parallel/test-http2-write-empty-string.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+
+const http2 = require('http2');
+
+const server = http2.createServer(function(request, response) {
+  console.log('responding to ' + request.url);
+
+  response.writeHead(200, {'Content-Type': 'text/plain'});
+  response.write('1\n');
+  response.write('');
+  response.write('2\n');
+  response.write('');
+  response.end('3\n');
+
+  this.close();
+});
+
+server.listen(0, common.mustCall(function() {
+  const client = http2.connect(`http://localhost:${this.address().port}`);
+
+  const headers = { ':path': '/' };
+  const req = client.request(headers).setEncoding('ascii');
+
+  let response = '';
+
+  req.on('response', (headers) => {
+    assert.strictEqual(200, headers[':status']);
+  });
+
+  req.on('data', function(chunk) {
+    response += chunk;
+  });
+
+  req.on('end', common.mustCall(function(a,b,c) {
+    assert.strictEqual('1\n2\n3\n', response);
```

Comment:
You need to close the client session.

Have you tried in moving `server.close()`  to `req.on('end')`, together with `server.destroy()`, as it's done everywhere else?